### ISSUE TITLE
build: darwin urbit-king release binary links against /nix-store

### ISSUE THREAD

**Issue by joemfb** [urbit/urbit#3900]
From the latest `release/next-vere`:

```
$ wget https://bootstrap.urbit.org/ci/931zq99kcrl031a3q2ifdrj53hi7r83k-urbit-v0.10.9-rc1-x86_64-darwin.tgz


$ tar xzvf 931zq99kcrl031a3q2ifdrj53hi7r83k-urbit-v0.10.9-rc1-x86_64-darwin.tgz 
x urbit-v0.10.9-rc1-x86_64-darwin/urbit
x urbit-v0.10.9-rc1-x86_64-darwin/urbit-king
x urbit-v0.10.9-rc1-x86_64-darwin/urbit-worker

$ otool -L urbit-v0.10.9-rc1-x86_64-darwin/urbit
urbit-v0.10.9-rc1-x86_64-darwin/urbit:
        /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1226.10.1)
        @rpath/CoreFoundation.framework/Versions/A/CoreFoundation (compatibility version 150.0.0, current version 1454.90.0)

$ otool -L urbit-v0.10.9-rc1-x86_64-darwin/urbit-worker 
urbit-v0.10.9-rc1-x86_64-darwin/urbit-worker:
        /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1226.10.1)
        @rpath/CoreFoundation.framework/Versions/A/CoreFoundation (compatibility version 150.0.0, current version 1454.90.0)

$ otool -L urbit-v0.10.9-rc1-x86_64-darwin/urbit-king 
urbit-v0.10.9-rc1-x86_64-darwin/urbit-king:
        /nix/store/qbbl5g6l5kx3zwjmkbcwflrzfrk6by0g-Libsystem-osx-10.12.6/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1226.10.1)
```

It looks like `urbit-king` needs to be fixed up with `install_name_tool` as well:

https://github.com/urbit/urbit/blob/86b59c67c26af2566731fc0eb119fefadea915e2/nix/pkgs/urbit/default.nix#L57-L63

BTW, is it worth trying to use this directly from nixpkgs? I believe we should have access to it already:

https://github.com/NixOS/nixpkgs/blob/11fc05583669cf3cb8c99dc6cb6f6ca7d93fa01d/pkgs/stdenv/darwin/portable-libsystem.sh#L1-L12

https://github.com/obsidiansystems/nixpkgs/blob/d6c0115cee9507b1908e7e1b4ce08995e429611d/pkgs/top-level/static.nix#L20-L32

Also worth noting, something seems off about our use of `install_name_tool`:

```
installing
error: install_name_tool: can't map file: /nix/store/d6zmzdc9g5qx09yj0760k0i0wmw0zz2a-urbit-static-0.10.9-rc1/bin (Invalid argument)
post-installation fixup
error: install_name_tool: can't map file: /nix/store/d6zmzdc9g5qx09yj0760k0i0wmw0zz2a-urbit-static-0.10.9-rc1/bin (Invalid argument)
patching script interpreter paths in /nix/store/d6zmzdc9g5qx09yj0760k0i0wmw0zz2a-urbit-static-0.10.9-rc1
```

(from https://hercules-ci.com/accounts/github/urbit/derivations/%2Fnix%2Fstore%2Fk89h8ll550xl9ijia44kcs5sy85s514i-urbit-static-0.10.9-rc1.drv/log?via-job=a8c3941d-e66a-4eaf-8148-6e2bc84342c8)

Is it just that we're invoking it against the directory as well as the files?

/cc @eglaysher

**Comment by brendanhay** [comment]
Definitely worth trying the `nixpkgs` version - I'll try that today.

**Comment by joemfb** [comment]
Fixed in #3901.
